x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Analyzer.java

```diff
@@ -341,6 +343,17 @@ else if (plan instanceof Aggregate) {
                     List<Expression> groupings = a.groupings();
                     List<Expression> newGroupings = new ArrayList<>();
                     AttributeMap<Expression> resolved = Expressions.aliases(a.aggregates());
+
+                    //if all we're selecting are functions or literals, even if we're giving them aliases.
+                    if(a.aggregates().stream().filter(s -> s.children().size()>0
```

Comment:
Please use `.forEachUp()`, you can find an example here: https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Analyzer.java#L638

Also please check your formatting regarding whitespaces around `(` and `)`, and around operators, e.g.: `size()>0` => `size() > 0`.

Reply:
not sure of the forEchUp usage.